src/libraries/System.Linq/src/System/Linq/Average.cs

```diff
@@ -102,6 +129,22 @@ public static double Average(this IEnumerable<long> source)
             }
         }
 
+        private static double Average(ReadOnlySpan<long> span)
```

Comment:
Are specializations like this worth the extra code when there's no vectorization? It seems like an `IList<T>` specialization might be more impactful here.

Reply:
<html>
<body>
<!--StartFragment-->

AverageInt64 | \main\corerun.exe | 32 | Array | 148.826 ns | 2.9388 ns | 1.00
-- | -- | -- | -- | -- | -- | --
AverageInt64 | \pr\corerun.exe | 32 | Array | 14.362 ns | 0.3140 ns | 0.10

<!--EndFragment-->
</body>
</html>